src/v/kafka/server/audit_log_manager.cc

```diff
@@ -0,0 +1,569 @@
+
+    /// Primitives for ensuring background work and toggling of switch w/ async
+    /// work occur in lock step
+    ss::gate _gate;
+    mutex _mutex;
+
+    /// Reference to audit manager so synchronization with its fibers may occur.
+    /// Supports pausing and resuming these fibers so the client can safely be
+    /// deallocated when no more work is concurrently entering the system
+    audit_log_manager* _audit_mgr;
+
+    /// audit_client and members necessary to pass to its constructor
+    std::unique_ptr<audit_client> _client;
+    cluster::controller* _controller;
+    client::configuration& _config;
+};
+
+audit_client::audit_client(
+  cluster::controller* controller, client::configuration& client_config)
+  : _send_sem(
+    config::shard_local_cfg().audit_client_max_buffer_size(),
+    "audit_log_producer_semaphore")
+  , _client(
+      config::to_yaml(client_config, config::redact_secrets::no),
+      [this](std::exception_ptr eptr) { return mitigate_error(eptr); })
+  , _controller(controller) {}
+
+ss::future<> audit_client::initialize() {
+    auto gh = _gate.hold();
+    while (!_as.abort_requested()) {
+        try {
+            co_await configure();
+            break;
+        } catch (...) {
+            /// Sleep, then try again
+        }
+        co_await ss::sleep_abortable(1s).handle_exception_type(
+          [](ss::sleep_aborted) {});
+    }
+}
+
+ss::future<> audit_client::configure() {
+    try {
+        auto config = co_await client::create_client_credentials(
+          *_controller,
+          config::shard_local_cfg(),
+          _client.config(),
+          audit_principal);
+        set_client_credentials(*config, _client);
+
+        auto const& store
+          = _controller->get_ephemeral_credential_store().local();
+        _has_ephemeral_credentials = store.has(store.find(audit_principal));
+
+        co_await set_auditing_permissions();
+        co_await create_internal_topic();
+
+        _client.config().retries.set_value(std::numeric_limits<size_t>::max());
+        vlog(klog.info, "Audit log subsystem initialized");
+    } catch (...) {
+        vlog(
+          klog.warn,
+          "Audit log subsystem failed to initialize: {}",
+          std::current_exception());
+        throw;
+    }
+}
+
+ss::future<> audit_client::set_auditing_permissions() {
+    /// Give permissions to create and write to the audit topic
+    security::acl_entry acl_create_entry{
+      audit_principal,
+      security::acl_host::wildcard_host(),
+      security::acl_operation::create,
+      security::acl_permission::allow};
+
+    security::acl_entry acl_write_entry{
+      audit_principal,
+      security::acl_host::wildcard_host(),
+      security::acl_operation::write,
+      security::acl_permission::allow};
+
+    security::resource_pattern audit_topic_pattern{
+      security::resource_type::topic,
+      model::kafka_audit_logging_topic,
+      security::pattern_type::literal};
```

Comment:
may just be simpler to do `security::acl_operation::all` rather than specific access for the audit topic